pkg/kubelet/kubelet.go

```diff
@@ -1347,6 +1350,33 @@ func (kl *Kubelet) Run(updates <-chan kubetypes.PodUpdate) {
 	if kl.kubeClient != nil {
 		// Start syncing node status immediately, this may set up things the runtime needs to run.
 		go wait.Until(kl.syncNodeStatus, kl.nodeStatusUpdateFrequency, wait.NeverStop)
+		// Start a loop that checks the internal node indexer cache for when a CIDR is applied
+		// and tries to update pod CIDR immediately. After pod CIDR is updated it fires off a runtime
+		// update and a node status update.
+		// This should significantly improve latency to ready node by updating pod CIDR, runtime status
+		// and node statuses ASAP.
+		// TODO(mtaufen): potentially generalize this to a fast "node ready" status update by
+		// factoring a readiness predicate out of setNodeReadyCondition in kubelet_node_status.go.
+		fastStatusUpdate := func() {
+				for {
+					time.Sleep(100 * time.Millisecond)
+					node, err := kl.GetNode()
+					if err != nil {
+						glog.Errorf(err.Error())
+						continue
+					}
+					if node.Spec.PodCIDR != "" {
+						if err := kl.updatePodCIDR(node.Spec.PodCIDR); err != nil {
+							glog.Errorf(err.Error())
+							continue
+						}
+						kl.updateRuntimeUp()
```

Comment:
I'm not sure if this thread safe.

Reply:
What exactly is not thread safe?
updateRuntimeUp and syncNodeStatus can be executed in parallel as they are executed in parallel even without this change.
I'm not sure if two updateRuntimeUp can be exectuted in parallel.
I'm also not sure if updatePodCIDR and syncNodeStatus can be executed in parallel.
Grabbing lock at top of the loop won't solve the problem. I could grab lock in updateRuntimeUp but if I use the same mutex as in syncNodeStatus then it might slow things down.
Should I use another lock in updatePodCIDR?

Reply:
Friendly ping:)

Reply:
done